Add basic support for polylines (i.e. open polygons)

Mesoptier opened the pull request:
As the title suggests, this PR adds support for the `<polyline>` SVG element. Since this element is so similar to the `<polygon>` SVG element, I figured it'd be better to add a prop to switch between the two, instead of adding a separate `<Polyline>` Mafs component.

I've only really added the prop and an example so far and have not updated any docs/tests yet. Let me know if you agree with this approach and I'll get on those docs and tests.

stevenpetryk commented:
I appreciate this! I actually think Polygon and Polyline should be separate components since they have mutually exclusive definitions and also ideally slightly different default behavior:

- Polygon must be a closed shape, and should have a fill by default.
- Polyline can be an open shape, and probably shouldn't have a fill by default.

I do think it'd be sensible for them to use the same underlying component/logic internally though :)

re: the tests, something I don't have documented at the moment is that the guide examples are snapshotted, so `pnpm test:e2e` will take care of cataloguing the `PolylineExample` and adding it to the test suite.

Thank you!

Mesoptier commented:
Fair enough! I've added a separate `<Polyline />` component and updated both components to use a shared internal `<PolyBase />` component. Also, `<Polyline />` defaults to `fillOpacity = 0`.

stevenpetryk commented:
Planning to release this with 0.16.0 :)
